### PR TITLE
Added support for Database

### PR DESCRIPTION
**Pull request by satakshigarg**
Terraform support for Database as a Service
Signed-off-by: satakshigarg <satakshi@civo.com>

### CODE REVIEW

File: civo/datasource_database.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+				Description:  "Count of nodes",
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The region of an existing Object Store",
+			},
```
Comment (RealHarshThakur) [inline civo/datasource_database.go:51]: Object Store -> Database. Probably check other files too.

File: civo/datasource_database.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			"network_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The network id of the Database",
+			},
+			"firewall_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The secret access key of the Database",
+			},
```
Comment (RealHarshThakur) [inline civo/datasource_database.go:61]: firewall ID of database.

File: civo/datasource_database.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		database, err := apiClient.FindDatabase(name.(string))
+		if err != nil {
+			return diag.Errorf("[ERR] failed to retrive Database: %s", err)
+		}
+
+		foundDatabase = database
+	}
+
+	if id, ok := d.GetOk("id"); ok {
+		log.Printf("[INFO] Getting the Database by name")
```
Comment (alejandrojnm) [inline civo/datasource_database.go:87]: it should be 
```
[INFO] Getting the Database by id
```

File: civo/resource_database.go [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+	database, err := apiClient.NewDatabase(config)
+	if err != nil {
+		return diag.Errorf("[ERR] failed to create Database: %s", err)
+	}
+
+	d.SetId(database.ID)
+
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{"pending"},
+		Target:  []string{"ready"},
```
Comment (alejandrojnm) [inline civo/resource_database.go:137]: `pending` and `ready` are the state of the database ??

**Review by alejandrojnm — COMMENTED** [review]
We need fix all the comment but in general look good to me

File: civo/resource_database.go (hunk 4 above)
Comment (satakshigarg) [reply civo/resource_database.go:137]: Yes, but I want a confirmation if "pending" and "ready"are correct for database or we have different state for database.

Comment (alejandrojnm) [reply civo/resource_database.go:137]: @vishalanarase can you confirm this ??

Comment (haardikdharma10) [reply civo/resource_database.go:137]: Yes that is correct

Comment (haardikdharma10) [reply civo/resource_database.go:137]: @satakshigarg change it to `Pending` and `Ready`

Comment (satakshigarg) [reply civo/resource_database.go]: Done !!

**Comment by alejandrojnm** [comment]
@satakshigarg @RealHarshThakur this is ready to review and merge, I run the ACC test locally and every test pass

**Comment by satakshigarg** [comment]
LGTM @alejandrojnm

**Pull request merged**